docs/debugging.md

```diff
@@ -0,0 +1,192 @@
+
+```bash title="Setup ledger tezos baking address to bake"
+$ tezos-signer setup ledger to bake for "ledger://stiff-sloth-grown-grouse/ed25519/0h/0h"
+Setting up the ledger:
+* Main chain ID: 'Unspecified' -> NetXLH1uAxK7CCh
+* Main chain High Watermark: 0 -> 0
+* Test chain High Watermark: 0 -> 0
+Authorized baking for address: tz1M3kiQUMPcTseo72i1twdiV4iTY5yiGNSz
+Corresponding full public key: edpkuNyLH57vRp2dyzeDSg2G7AarZcp29DQnkQaXrSNdkRkAE4B1ZA
+```
+
+## Testing signatures. 
+
+Here are some sample signing commands for different operations. These operations were taken from an Ithaca chain. 
+
+```bash title="Signing an example Endorsement for testing purposes (Magic byte 0x13)"
+$ tezos-client sign bytes 0x137a06a7703e47ad79416e7bdceb65156abdbcfd4b1237caf488cbfc39d88c836840bf0bc81509b9002513b8000000008c41b04d3a8648732fb22507447214aa698a235862f30b956e36a37a7d36eb7c for octez_signer_remote_ledger
+Signature: edsigu66k8pYbKvhRbQizVL41XApHz9y6FSj4E6CowvWUsD13j7BDQJAJMh3id4LFhqsip6AhnpcYhRmvMyV4enAnyvEHbTiVZZ
+```
+
+```bash title="Signing an example Pre-Endorsement for testing purposes (Magic byte 0x12)"
+$ tezos-client sign bytes 0x127a06a7703e47ad79416e7bdceb65156abdbcfd4b1237caf488cbfc39d88c836840bf0bc81409b9002513b8000000008c41b04d3a8648732fb22507447214aa698a235862f30b956e36a37a7d36eb7c for octez_signer_remote_ledger
+Signature: edsigtaMbMAePuHrEi6cHAc8qKJjrdF9LWdewBJrmrYZTKWAyV1JYADySGovoqkuvwY68UwBxxyj3MJ2Ft7pNAKhbhc6gP4VP8p
+```
+
+```bash title="Signing an example block for testing purposes (Magic byte 0x11)"
+tezos-client sign bytes 0x117a06a770002397ee0c68e4ef4a35e2f768012c2f3b560bb80eb8849327696b6843e20842ca5e01345f000000006270c6650469204194e88ec93cdc7ee5d0aa42908ad49e4e0fb7242d406e73d3a48528d4af00000021000000010200000004002397ee0000000000000004ffffffff000000040000000076222f1388f0a7d6b53b96fde78a2b79b50a19282c48eb003cbdffb6b3669e3897440e9db3d73900616e3f720e162588071444071312d967a7d38f7ea23a480c0000000061fed54022b203000000 for octez_signer_remote_ledger
+```
+
+The ledger device will prompt you to confirm sharing the public key, and after that, `tezos-signer` will add a new entry to its configuration files in ~/.tezos-signer`
+
+#### configure tezos-client to use tezos-signer
+
+```bash title="tezos-client creating a new account using a remote signer service"
+tezos-client import secret key octez_signer_remote_ledger http://localhost:6732/tz1M3kiQUMPcTseo72i1twdiV4iTY5yiGNSz
+Tezos address added: tz1M3kiQUMPcTseo72i1twdiV4iTY5yiGNSz
+```
+:::info
+`tezos-client` has a quirk where we pass the URI http://localhost:6732/tz1M3kiQUMPcTseo72i1twdiV4iTY5yiGNSz to the command, but if you do a curl on that URL, the `tezos-signer` will return a 404. The `tezos-client` rewrites the URL to include the `/keys/` slug. If you want to get the public key for a PKH from the remote signer, you can use curl like this: 
+
+```bash title="fetching public key from remote signer using curl"
```

Comment:
The F in fetching should be capitalized to match all other sections like this

Reply:
fixed